metadata-ingestion/src/datahub/sdk/mlmodel.py

```diff
@@ -0,0 +1,328 @@
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Any, Dict, List, Optional, Type, Union
+
+from typing_extensions import Self
+
+from datahub.emitter.mce_builder import DEFAULT_ENV
+from datahub.metadata.schema_classes import (
+    AspectBag,
+    MLHyperParamClass,
+    MLMetricClass,
+    MLModelPropertiesClass,
+    VersionPropertiesClass,
+    VersionTagClass,
+)
+from datahub.metadata.urns import MlModelUrn, Urn, VersionSetUrn
+from datahub.sdk._shared import (
+    DomainInputType,
+    HasDomain,
+    HasInstitutionalMemory,
+    HasOwnership,
+    HasPlatformInstance,
+    HasTags,
+    HasTerms,
+    LinksInputType,
+    OwnersInputType,
+    TagsInputType,
+    TermsInputType,
+    make_time_stamp,
+    parse_time_stamp,
+)
+from datahub.sdk.entity import Entity, ExtraAspectsType
+
+
+class MLModel(
+    HasPlatformInstance,
+    HasOwnership,
+    HasInstitutionalMemory,
+    HasTags,
+    HasTerms,
+    HasDomain,
+    Entity,
+):
+    __slots__ = ()
+
+    @classmethod
+    def get_urn_type(cls) -> Type[MlModelUrn]:
+        return MlModelUrn
+
+    def __init__(
+        self,
+        id: str,
+        platform: str,
+        version: Optional[str] = None,
+        aliases: Optional[List[str]] = None,
+        platform_instance: Optional[str] = None,
+        env: str = DEFAULT_ENV,
+        name: Optional[str] = None,
+        description: Optional[str] = None,
+        display_name: Optional[str] = None,
+        training_metrics: Optional[Union[List[MLMetricClass], Dict[str, Any]]] = None,
```

Comment:
extract these into TrainingMetricsInputType, similar to how it's done elsewhere

Reply:
done